erts/emulator/beam/erl_printf_term.c

```diff
@@ -185,7 +183,36 @@ static int is_printable_ascii(byte* bytep, Uint bytesize, Uint bitoffs)
     return 1;
 }
 
-/* print a atom doing what quoting is necessary */
+/*
+ * Helper function for print_atom_name(). Not generally useful.
+ */
+static ERTS_INLINE int latin1_char(int c1, int c2)
+{
+    if ((c1 & 0x80) == 0) {
+        /* Plain old 7-bit ASCII. */
+        return c1;
+    } else if ((c1 & 0xE0) == 0xC0) {
+        /* Unicode code points from 0x80 through 0xFF. */
+        ASSERT((c2 & 0xC0) == 0x80);
+        return (c1 & 0x1F) << 6 | (c2 & 0x3F);
```

Comment:
Code and comment still doesn't match.
The code decodes a full 2-byte code point, that is 5+6=11 bits up to 0x7FF.

To only decode the "latin1" interval up to 0xFF the three most significant payload bits must be zero:

`} else if ((c1 & 0xFC) == 0xC0) { `

Reply:
> Code and comment still doesn't match.

Thanks. I will update the comment. (The characters 0x100 - 0x7FF will be handled by the caller.)

Reply:
Sorry that was my fault, forgot about higher bits 😅 👍